### PR TITLE
techdebt: [CI-9497]: Integrated SLSA external parameters

### PR DESCRIPTION
**Pull request by sahithibanda01**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52569)

### CODE REVIEW

File: 879-pipeline-ci-commons/src/main/java/io/harness/beans/provenance/BuildMetadata.java [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
 
-import io.harness.annotations.dev.HarnessTeam;
-import io.harness.annotations.dev.OwnedBy;
+import io.harness.pms.yaml.ParameterField;
 
+import java.util.Map;
 import lombok.Builder;
 import lombok.Data;
 
 @Data
```
Comment (harshjain12) [inline 879-pipeline-ci-commons/src/main/java/io/harness/beans/provenance/BuildMetadata.java:9]: @value

File: 879-pipeline-ci-commons/src/main/java/io/harness/beans/provenance/ProvenanceGenerator.java [hunk 2]
```diff
@@ -36,17 +38,22 @@ public ProvenancePredicate buildProvenancePredicate(ProvenanceBuilderData data)
                                                      .pipelineExecutionId(data.getPipelineExecutionId())
                                                      .pipelineIdentifier(data.getPipelineIdentifier())
                                                      .build())
+                             .externalParameters(ExternalParameters.builder()
```
Comment (harshjain12) [inline 879-pipeline-ci-commons/src/main/java/io/harness/beans/provenance/ProvenanceGenerator.java:41]: passExternal params here directly

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/DockerStep.java [hunk 3]
```diff
@@ -101,11 +110,60 @@ private void populateArtifact(ArtifactMetadata artifactMetadata, StepElementPara
...
-  private void populateProvenanceInStepOutcome(Ambiance ambiance, StepArtifactsBuilder stepArtifactsBuilder) {
+  private void populateProvenanceInStepOutcome(
+      Ambiance ambiance, StepArtifactsBuilder stepArtifactsBuilder, StepElementParameters stepParameters) {
     String accountId = AmbianceUtils.getAccountId(ambiance);
     if (!featureFlagService.isEnabled(FeatureName.SSCA_SLSA_COMPLIANCE, accountId)) {
       return;
     }
+
+    // fetching CodeMetadata of ExternalParameters
+    OptionalSweepingOutput optionalSweepingOutput =
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/DockerStep.java:121]: refactor this method

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/DockerStep.java [hunk 4]
```diff
@@ -101,11 +110,60 @@ private void populateArtifact(ArtifactMetadata artifactMetadata, StepElementPara
     }
   }
 
-  private void populateProvenanceInStepOutcome(Ambiance ambiance, StepArtifactsBuilder stepArtifactsBuilder) {
+  private void populateProvenanceInStepOutcome(
+      Ambiance ambiance, StepArtifactsBuilder stepArtifactsBuilder, StepElementParameters stepParameters) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/DockerStep.java:114]: Pass DockerStepInfo dockerStepInfo

**Review by harshjain12 — CHANGES_REQUESTED** [review]

File: 879-pipeline-ci-commons/src/main/java/io/harness/beans/provenance/TriggerMetadata.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package io.harness.beans.provenance;
+
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+import lombok.Data;
+import lombok.Value;
+
+@Data
```
Comment (harshjain12) [inline 879-pipeline-ci-commons/src/main/java/io/harness/beans/provenance/TriggerMetadata.java:9]: remove @Data
@value is sufficient

**Comment by sahithibanda01** [comment]
trigger messagecheck

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/mapper/NGTriggerElementMapper.java [hunk 6]
```diff
@@ -226,12 +226,12 @@ public void copyEntityFieldsOutsideOfYml(NGTriggerEntity existingEntity, NGTrigg
...
     }
     if (newEntity.getType() == MULTI_REGION_ARTIFACT) {
-      /* MultiRegionArtifact triggers need different handling here, since we can't just copy the list of BuildMetadata
-      from the previously existing entity (e.g.: The number of artifacts it is listening for could have changed in
-      the update). So we only copy the previously existing signatures to `ngTriggerEntity.metadata.signatures`.
-      These will be used to unsubscribe from the previously subscribed poolingDocuments.
-      At a later step (in NGTriggerServiceImpl:stampPollingInfoForMultiArtifactTrigger) we reset the signatures
-      with the actual new signatures for this trigger. */
+      /* MultiRegionArtifact triggers need different handling here, since we can't just copy the list of
+      RunDetailsMetadata from the previously existing entity (e.g.: The number of artifacts it is listening for could
```
Comment (inderschera) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/mapper/NGTriggerElementMapper.java:230]: let's undo this change.

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java [hunk 7]
```diff
@@ -483,7 +483,7 @@ public void stampPollingInfoForMultiArtifactTrigger(
       ngTriggerEntity.getMetadata().setMultiBuildMetadata(Collections.emptyList());
       ngTriggerEntity.getMetadata().setSignatures(Collections.emptyList());
     } else {
-      // Copy each pollingDocId to the corresponding BuildMetadata in Triggers metadata.
+      // Copy each pollingDocId to the corresponding RunDetailsMetadata in Triggers metadata.
```
Comment (inderschera) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java:486]: let's undo this change.

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/generator/PollingItemGenerator.java [hunk 8]
```diff
@@ -57,7 +57,7 @@ default PollingItem.Builder getBaseInitializedPollingItem(
     String pollingDocId;
     if (type == MULTI_REGION_ARTIFACT) {
       /* For MultiRegionArtifact triggers, we need to fetch signature and pollingDocId from `buildTriggerOpsData`,
-      because the trigger's metadata itself contains a list of BuildMetadata, so we don't know which element of the
+      because the trigger's metadata itself contains a list of RunDetailsMetadata, so we don't know which element of the
```
Comment (inderschera) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/generator/PollingItemGenerator.java:60]: let's undo this change

**Comment by sahithibanda01** [comment]
trigger uts

**Comment by sahithibanda01** [comment]
trigger uts

**Comment by sahithibanda01** [comment]
trigger uts

**Comment by harshjain12** [comment]
uts are failing

**Pull request merged**